File: CLIPrestashopConnector/Dtos/ToCliDto.cs
using System;
using CLIPrestashopConnector.Models;
using Microsoft.EntityFrameworkCore.Storage.ValueConversion;

namespace CLIPrestashopConnector.Dtos
{
    public class ToCliDto
    {
        public long Id { get; set; } 
        public long Id_T_Famille { get; set; } 
        public string Reference { get; set; } 
        public bool Force { get; set; } = false;
        public bool AssociatedAddress { get; set; } = false;
        public bool AssociatedCartRule { get; set; } = false;
        public bool AssociatedLegacyImages { get; set; } = false;
        public string Image { get; set; } = string.Empty;
        public long DefaultImageId { get; set; } 
        public byte[] FactureData { get; set; }
        public bool OnlyErrors { get; set; } = false;
        public bool OnlyNewSync { get; set; } = false;
        public bool ImportStock { get; set; } = false;
        public bool DeleteBeforeImport { get; set; } = false;
        public DateTime? UpdatedDateFrom { get; set; } = null;
        public List<long> ToDeleteImages { get; set; }
        public List<ImageData> ToAddImages { get; set; }
        public int Number { get; set; } = 0;
        // pour gérer les produits en masse
        public List<long> Ids { get; set; }
            public string CurrentMessage { get; set; }= string.Empty;
    public string NewMessage { get; set; }= string.Empty;
        public ToCliDto()
        {
            ToDeleteImages = new List<long>();
            ToAddImages = new List<ImageData>();
            Ids = new List<long>();
        }
    }
}

File: CLI/Model/ToCliDto.vb
Public Class ToCliDto
    Public Id As Long
    Public Id_T_Famille As Long
    Public Reference As String = ""
    Public Force As Boolean = False
    Public AssociatedAddress As Boolean = False
    Public AssociatedCartRule As Boolean = False
    Public AssociatedLegacyImages As Boolean = False
    Public OnlyErrors As Boolean = False
    Public OnlyNewSync As Boolean = False
    Public UpdatedDateFrom As Date? = Nothing
    Public ImportStock As Boolean = False
    Public DeleteBeforeImport As Boolean = False
    Public Image As String = ""
    Public FactureData As Byte()
    Public DefaultImageId As Long
    Public ToDeleteImages As List(Of Long)
    Public ToAddImages As List(Of ImageData)
    Public Number As Int16
    'Pour gérer les produits en masse
    Public Ids As List(Of Long)
    Public CurrentMessage As String = ""
    Public NewMessage As String = ""
End Class
